doc/sphinxman/source/build_planning.rst

```diff
@@ -1447,3 +1447,53 @@ Ways to refer to |PSIfour| in text, in order of decreasing goodness:
 
   * **NOT** ``PSI4`` or ``PSI``
 
+
+.. _`faq:localaddon`:
+
+How to use a local Add-On repository in the Psi4 build
+------------------------------------------------------
+
+For each Add-On, |PSIfour| pulls source from a specific online Git
+repository and a specific tag/branch/commit in it. This ensures success
+of the |PSIfour| build, reproducibility of the runtime results, and
+freedom for continued upstream development. Sometimes, you're the one
+doing that development, and you need the CMake superbuild to pull source
+from a local path rather than the approved codeset.
+
+Find the ``CMakeLists.txt`` governing the target Add-On in
+:source:`external` and make changes analogous to the below::
+
+    #GIT_REPOSITORY https://github.com/jturney/ambit
+    #GIT_TAG 1.0
+    DOWNLOAD_COMMAND ""
+    SOURCE_DIR "/path/to/ambit-directclone"
+
+If you're changing the |PSIfour| repo codebase between compiles, there's
+nothing more to do as CMake will handle the code rebuild deps for you.
+
+If you're changing the local Add-On repo codebase between compiles,
+CMake *does not* know when ``libaddon.a/so/dylib`` needs rebuilding. It
+is recommended that the |PSIfour| build be initially configured with
+``-DBUILD_SHARED_LIBS=ON`` (easier to notice changes). And to trigger
+Add-On library rebuild, ``rm -rf {objdir}/external/upstream/addon/``
+and ``rm -rf {objdir}/stage/{prefix}/share/cmake/AddOn``. This should
+re-clone the Add-On, rebuild and install it, rebuild any parts of
+|PSIfour| that interface to it, and relink the main ``core.so``.
+If you're modifying the Add-On's file or directory structure, be
+smart and ``rm`` all traces of it within ``{objdir}/stage/{prefix}``,
+especially any ``*.pyc`` files.
```

Comment:
Ditto here `make clean python`.

Reply:
For this and previous, yes, looks like we can define external commands that will forcibly clean up whatever. They will be whatever names we choose, though, and not have any relation to actual `make clean` (in fact, cmake isn't big on "clean" b/c it's only really defined for GnuMake, not Ninja, etc. https://cmake.org/pipermail/cmake/2016-December/064695.html)

I tend to prefer the direct `rm` approach, since at this detail of dev one should know where files are going. But I'll certainly entertain seconds to the proposal.

Reply:
I like the rm approach as well. I worry a bit that while nice, this could get lost in the noise of features if we do implement it. However, it is handy on those off cases in which you do need it.